pkg/service/credential/storage.go

```diff
@@ -223,16 +227,55 @@ func (cs *Storage) StoreCredential(ctx context.Context, request StoreCredentialR
 	return cs.storeCredential(ctx, request, credentialNamespace)
 }
 
-func (cs *Storage) StoreStatusListCredentialTx(ctx context.Context, request StoreCredentialRequest, tx storage.Tx) error {
-	wc, err := cs.getStoreCredentialWriteContext(request, statusListCredentialNamespace)
+func (cs *Storage) StoreStatusListCredentialTx(ctx context.Context, request StoreCredentialRequest, statusPurpose statussdk.StatusPurpose, tx storage.Tx) error {
+	if !request.IsValid() {
+		return util.LoggingNewError("store request request is not valid")
+	}
+
+	// transform the credential into its denormalized form for storage
+	storedCredential, err := buildStoredCredential(request)
 	if err != nil {
-		return errors.Wrap(err, "could not get stored credential write context")
+		return errors.Wrap(err, "could not build stored credential")
```

Comment:
While you're here :) 
```suggestion
		return errors.Wrap(err, "building stored credential")
```